src/URI.js

```diff
@@ -524,6 +524,8 @@
     // what's left must be the path
     parts.path = string;
 
+    URI.basicValidation(parts);
```

Comment:
I've created and called the method this way because I wanted to put some validation there that isn't just about a specific part of the url but about a combination such as protocol, hostname, path.